src/components/Map/MainMap.js

```diff
@@ -30,6 +31,18 @@ class MainMap extends Component {
 
     componentDidMount() {
         const { geoid, dataset } = this.props;
+        const state = geoid.slice(0, 2);
+
+        // fetch('https://covid-scenario-dashboard.s3.amazonaws.com/json-files/statsForMap.json')
+        //     .then(response => response.json())
+        //     .then(data => this.setState({statsForCounty: data[state]}));
+
+        fetchJSON(geoid)
```

Comment:
I'm not familiar with the old wait promise/resolve works but if you want to do it with async/await you need to make componentDidMount async and then use `await` with fetchJSON.

Reply:
Note: then is call after the API call, but because `this.initializeMap(geoid, dataset)` is the next line after the API method, it also gets called immediately.

Reply:
```
// @lily, we might want to look into whether it's okay to have an
// async method in componentDidMount here. I *believe* it's okay
// but FYI in case something weird happens here.
async componentDidMount() {
    const { geoid, dataset } = this.props;
    const state = geoid.slice(0, 2);
    
    try {
        const statsForMap = await fetchJSON(geoid);
        this.setState({statsForCounty: statsForMap[state]};
        this.initializeMap(geoid, dataset)
    } catch (e) {
        console.log('Fetch was problematic: ' + e.message)
    } finally {
        // If we need to initializeMap regardless of fetch success,
        // we can call in the finally clause which occurs once after
        // try and catch
        this.initializeMap(geoid, dataset)
    }
};
```